### PR TITLE
Update the uninstalling Metering documentation.

### PR DESCRIPTION
**Pull request by timflannagan**

### CODE REVIEW

File: modules/metering-uninstall.adoc [hunk 1]
```diff
@@ -3,25 +3,33 @@
...
+= Uninstalling the Metering Operator from {product-title}
 
-You can remove metering from your cluster.
+The following guide provides details on removing the Metering Operator and its components.
 
-.Prerequisites
+[NOTE]
+====
+Metering does not manage or delete any S3 bucket data. If you used Amazon S3 for storage, any S3 buckets used to store metering data must be manually cleaned up.
+====
```
Comment (timflannagan1) [inline modules/metering-uninstall.adoc:13]: I moved this note further up in the document as it felt out of place mentioning this towards the bottom.

**Comment by timflannagan1** [comment]
It looks like Github may still be having some issues as I'm not seeing the last force-push get rendered yet.

File: modules/metering-uninstall.adoc [hunk 2]
```diff
@@ -3,25 +3,60 @@
...
 
-. In the dialog box, click *Remove* to uninstall metering.
+== Uninstalling metering CRDs
+
+Once the Metering Operator has been uninstalled and the `openshift-metering` namespace has been deleted, the metering CRDs will still exist in the cluster.
 
 [NOTE]
 ====
-Metering does not manage or delete any S3 bucket data. If you used Amazon S3 for storage, any S3 buckets used to store metering data must be manually cleaned up.
+Deleting the metering CRDs will disrupt any additional metering installations. Ensure no other installations exist before proceeding.
```
Comment (timflannagan1) [inline modules/metering-uninstall.adoc:47]: Open to suggestions on this language. Not entirely sure what's the best way to convey, "don't delete CRDs if there are any additional metering installations in other namespaces".

File: modules/metering-uninstall.adoc [hunk 3]
```diff
@@ -3,25 +3,60 @@
...
+[NOTE]
+====
+Metering does not manage or delete any S3 bucket data. If you used Amazon S3 for storage, any S3 buckets used to store metering data must be manually cleaned up.
+====
+
+== Removing the Metering Operator
+
+You can remove the Metering Operator by following the documentation on xref:../../operators/olm-deleting-operators-from-cluster.adoc#olm-deleting-operators-from-a-cluster[deleting Operators from a cluster].
+
+== Uninstalling metering
```
Comment (timflannagan1) [inline modules/metering-uninstall.adoc:19]: We may want to expose a CLI and console (UI) procedure.

File: modules/metering-uninstall.adoc [hunk 4]
```diff
@@ -3,25 +3,60 @@
...
+. You have uninstalled Metering following the documentation listed in the previous step.
 
-To remove metering:
+. Procedure
 
-. In the {product-title} web console, click *Operators* -> *Installed Operators*.
+. Use the following command to remove all resources created by the Metering Operator:
++
+----
+$ oc --namespace openshift-metering delete meteringconfigs.metering.openshift.io openshift-metering
```
Comment (pruan-rht) [inline modules/metering-uninstall.adoc:32]: @timflannagan1 perhaps we can shorten this to `oc delete meteringconfigs <YOUR_METERINGCONFIG_NAME> -n openshift-metering.  Also,

File: modules/metering-uninstall.adoc [hunk 5]
```diff
@@ -3,25 +3,60 @@
...
+. You have deleted the `MeteringConfig` custom resource.
+
+. You have deleted the `openshift-metering` namespace.
+
+. Procedure
+
+. Delete the remaining metering CRDs be running the following command:
++
+----
+$ oc get crd -o name | grep "metering.openshift.io" | xargs oc delete
```
Comment (timflannagan1) [inline modules/metering-uninstall.adoc:61]: I, unfortunately, don't have a way of cleanly deleting these resources without using tooling like `xargs`.

File: modules/metering-uninstall.adoc (hunk 4 above)
Comment (timflannagan1) [reply modules/metering-uninstall.adoc:32]: Yeah there's some potential variability with the MeteringConfig custom resource name. I think the doc standard is to use something like the following:

```bash
$ oc delete meteringconfig <1> meteringconfigs

<1> The name of the MeteringConfig custom resource present in the openshift-metering namespace.
```

File: modules/metering-uninstall.adoc [hunk 6]
```diff
@@ -3,25 +3,62 @@
 // * metering/metering-uninstall.adoc
 
 [id="metering-uninstall_{context}"]
-= Uninstalling metering from {product-title}
+= Uninstalling the Metering Operator from {product-title}
 
-You can remove metering from your cluster.
+The following guide provides details on removing the Metering Operator and its components.
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:8]: Suggest rewording this if you want to be specific that you're removing the Metering Operator from your cluster:

You can remove the Metering Operator and associated metering components from your cluster.

File: modules/metering-uninstall.adoc [hunk 7]
```diff
@@ -3,25 +3,62 @@
...
 
-.Prerequisites
+[NOTE]
+====
+Metering does not manage or delete any S3 bucket data. If you used Amazon S3 for storage, any S3 buckets used to store metering data must be manually cleaned up.
+====
+
+== Removing the Metering Operator
+
+You can remove the Metering Operator by following the documentation on xref:../operators/olm-deleting-operators-from-cluster.adoc#olm-deleting-operators-from-a-cluster[deleting Operators from a cluster].
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:17]: IIRC, we don't use xrefs within a module. I think it would be best to include the procedure for removing the Metering Operator with metering-specific example commands and output.

Comment (timflannagan1) [reply modules/metering-uninstall.adoc:17]: That sounds reasonable to me, but I would imagine there's still value in still linking to that documentation, even if we aim to emulate those steps with a metering-specific workflow.

File: modules/metering-uninstall.adoc [hunk 8]
```diff
@@ -3,25 +3,62 @@
...
+. Prerequisites
 
-.Procedure
+. You have uninstalled Metering following the documentation listed in the previous step.
 
-To remove metering:
+. Procedure
 
-. In the {product-title} web console, click *Operators* -> *Installed Operators*.
+. Use the following command to remove all resources created by the Metering Operator:
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:29]: s/Use the following command to/Remove all resources created by the Metering Operator:

**Comment by timflannagan1** [comment]
PTAL @openshift/team-documentation

File: modules/metering-uninstall.adoc [hunk 9]
```diff
@@ -3,25 +3,63 @@
 // * metering/metering-uninstall.adoc
 
 [id="metering-uninstall_{context}"]
-= Uninstalling metering from {product-title}
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:6]: Should this stay as "Uninstalling metering from OCP" since you are removing the Metering Operator _and_ all metering components? So it's more than only uninstalling the Operator?

File: modules/metering-uninstall.adoc [hunk 10]
```diff
@@ -3,25 +3,63 @@
...
+Metering does not manage or delete any S3 bucket data. If you used Amazon S3 for storage, any S3 buckets used to store metering data must be manually cleaned up.
+====
+
+== Removing the Metering Operator
+
+You can remove the Metering Operator by following the documentation on xref:../operators/olm-deleting-operators-from-cluster.adoc#olm-deleting-operators-from-a-cluster[deleting Operators from a cluster].
+
+== Uninstalling metering
+
+To uninstall metering, you must remove its custom resource and delete the `openshift-metering` namespace.
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:21]: s/remove its custom resource/remove the MeteringConfig resource/

File: modules/metering-uninstall.adoc [hunk 11]
```diff
@@ -3,25 +3,63 @@
...
 
 .Prerequisites
 
-* Metering must be installed.
+* You have uninstalled Metering following the documentation listed in the previous step.
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:25]: Is this prereq referring to deleting the Metering Operator from the cluster? If so, I suggest rewording this:

* The Metering Operator is removed from your cluster.

File: modules/metering-uninstall.adoc [hunk 12]
```diff
@@ -3,25 +3,63 @@
...
++
+----
+$ oc --namespace openshift-metering delete meteringconfigs <1>
+----
++
+<1> The name of the MeteringConfig custom resource that is present in the openshift-metering namespace.
++
 
-. In the {product-title} web console, click *Operators* -> *Installed Operators*.
+. Once that command has completed, verify all of the pods listed in the `openshift-metering` namespace have been deleted, and run the following command:
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:38]: s/Once the previous step is complete, verify that all Pods in the `openshift-metering` namespace are deleted.

(New step)
. Delete the `openshift-metering` namespace:

File: modules/metering-uninstall.adoc [hunk 13]
```diff
@@ -3,25 +3,63 @@
...
++
+----
+$ oc delete namespace openshift-metering
+----
 
-. Find the Metering Operator and click the {kebab} menu. Select *Uninstall Operator*.
+== Uninstalling metering CRDs
 
-. In the dialog box, click *Remove* to uninstall metering.
+Once the Metering Operator has been uninstalled and the `openshift-metering` namespace has been deleted, the metering CRDs will still exist in the cluster.
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:46]: Suggest changing this to present tense:

Once the Metering Operator is uninstalled and the `openshift-metering` namespace is deleted, the metering CRDs remain in the cluster.

File: modules/metering-uninstall.adoc (hunk 2 above)
Comment (lbarbeevargas) [reply modules/metering-uninstall.adoc:50]: Wording suggestion:

Deleting the metering CRDs disrupts any additional metering installations in other namespaces. Ensure that there are no other metering installations before proceeding.

File: modules/metering-uninstall.adoc [hunk 14]
```diff
@@ -3,25 +3,63 @@
...
-Metering does not manage or delete any S3 bucket data. If you used Amazon S3 for storage, any S3 buckets used to store metering data must be manually cleaned up.
+Deleting the metering CRDs will disrupt any additional metering installations. Ensure no other installations exist before proceeding.
 ====
+
+.Prerequisites
+
+* You have deleted the `MeteringConfig` custom resource.
+
+* You have deleted the `openshift-metering` namespace.
+
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:58]: Suggest changing these prereqs to present tense:

* The MeteringConfig custom resource is deleted.
* The `openshift-metering` namespace is deleted.

File: modules/metering-uninstall.adoc [hunk 15]
```diff
@@ -3,25 +3,63 @@
...
+.Procedure
+
+. Delete the remaining metering CRDs by running the following command:
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:61]: Change this to a bullet point since there is only one step in this procedure.

* Delete the remaining metering CRDs:

File: modules/metering-uninstall.adoc (hunk 9 above)
Comment (pruan-rht) [reply modules/metering-uninstall.adoc:6]: @lbarbeevargas @timflannagan1 actually the metering operator is isolated by namespace...so this should read `Uninstalling metering from your namespace`

**Comment by pweil-** [comment]
@timflannagan1 - bump.  Looks like this is awaiting minor updates and we can call the PR and jira complete.

File: modules/metering-uninstall.adoc (hunk 11 above)
Comment (timflannagan1) [reply modules/metering-uninstall.adoc:25]: The prerequisite here is that a user has already removed the metering-related OLM resources and not the metering installation itself.

**Comment by timflannagan1** [comment]
@lbarbeevargas finally got back to this PR about ~month if you don't mind taking another look. I addressed most of the review comments you had.

**Comment by timflannagan1** [comment]
Hmm, it looks like I may have pushed to the wrong branch. I'll try and fix this without having to open a separate PR.

File: modules/metering-uninstall.adoc [hunk 16]
```diff
@@ -3,25 +3,66 @@
...
+Metering does not manage or delete any S3 bucket data. If you used Amazon S3 for storage, any S3 buckets used to store metering data must be manually cleaned up.
+====
+
+== Removing the Metering Operator
+
+You can remove the Metering Operator by following the documentation on xref:../operators/olm-deleting-operators-from-cluster.adoc#olm-deleting-operators-from-a-cluster[deleting Operators from a cluster].
+
+== Uninstalling metering
+
+To uninstall metering, you must remove the MeteringConfig resource and delete the `openshift-metering` namespace.
```
Comment (timflannagan1) [inline modules/metering-uninstall.adoc:21]: @lbarbeevargas is there a doc standard as to when to use resource vs. custom resource?

Comment (timflannagan1) [reply modules/metering-uninstall.adoc:21]: Also, do you know if MeteringConfig -> `MeteringConfig` here?

**Comment by timflannagan1** [comment]
Alright figured out my branch issue and re-pushed. It looks like everything is back to normal now.

File: modules/metering-uninstall.adoc [hunk 17]
```diff
@@ -3,25 +3,66 @@
...
+. Remove all resources created by the Metering Operator:
++
+----
+$ oc --namespace openshift-metering delete meteringconfigs <1>
+----
++
+<1> The name of the MeteringConfig resource that is present in the openshift-metering namespace.
++
+
+. Once the previous step is complete, verify that all Pods in the openshift-metering namespace are deleted.
```
Comment (lbarbeevargas) [inline modules/metering-uninstall.adoc:38]: Is there a command to check that all the pods are deleted?

Comment (timflannagan1) [reply modules/metering-uninstall.adoc:38]: Yeah, you could run `$ oc --namespace openshift-metering get pods` and check that the no Pods get returned or all the Pods are reporting a terminating state.

File: modules/metering-uninstall.adoc (hunk 16 above)
Comment (lbarbeevargas) [reply modules/metering-uninstall.adoc:21]: @timflannagan1 I'm checking on the question regarding resource vs custom resource. Based on the [Installing metering](https://docs.openshift.com/container-platform/4.5/metering/metering-installing-metering.html) docs, I'm predominantly seeing "MeteringConfig resource." 

Also, with the exception of [About configuring metering](https://docs.openshift.com/container-platform/4.5/metering/configuring_metering/metering-about-configuring.html), MeteringConfig does not have backticks.

**Comment by lbarbeevargas** [comment]
CP back to 4.3+

**Comment by lbarbeevargas** [comment]
This work was completed in [PR-24969](https://github.com/openshift/openshift-docs/pull/24969).

https://docs.openshift.com/container-platform/4.5/metering/metering-uninstall.html
